src/Workspaces/Core/Portable/EmbeddedLanguages/VirtualChars/VirtualCharSequence.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars
+{
+    /// <summary>
+    /// Represents the individual characters that raw string token represents (i.e. with escapes collapsed).  
+    /// The difference between this and the result from token.ValueText is that for each collapsed character
+    /// returned the original span of text in the original token can be found.  i.e. if you had the
+    /// following in C#:
+    ///
+    /// "G\u006fo"
+    ///
+    /// Then you'd get back:
+    ///
+    /// 'G' -> [0, 1) 'o' -> [1, 7) 'o' -> [7, 1)
+    ///
+    /// This allows for embedded language processing that can refer back to the users' original code
```

Comment:
>users [](start = 80, length = 5)

Found the typo in this PR (`user's`). Did I pass the M&M's test?

Reply:
M&M test?

Reply:
It's a famous anecdote by a rock band that would include a clause in the contract with venues: "a bowl of M&M chocolates must be provided, but with no M&M's of a given color". This was a test to see that the venue had read the contract, since it included important safety information.

---
In reply to: [262241758](https://github.com/dotnet/roslyn/pull/33834#discussion_r262241758) [](ancestors = 262241758)